lib/absinthe/schema.ex

```diff
@@ -100,6 +100,44 @@ defmodule Absinthe.Schema do
 
   end
   ```
+
+  ## Default Resolver
+
+  By default, if a `resolve` function is not provided for a field, Absinthe
+  will attempt to extract the value of the field using `Map.get/2` with the
+  (atom) name of the field.
+
+  You can change this behavior by setting your own custom default resolve
+  function in your schema. For example, given we have a field, `name`:
+
+  ```
+  field :name, :string
+  ```
+
+  And we're trying to extract values from a horrible backend API that gives us
+  maps with uppercase (!) string keys:
+
+  ```
+  %{"NAME" => "A name"}
+  ```
+
+  Here's how we could set our custom resolver to expect those keys:
+
+  ```
+  default_resolve fn
+    _, %{source: source, definition: %{name: name}} ->
+      case source do
+        # Only from sources that are maps
+        %{} ->
+          {:ok, Map.get(source, String.upcase(name))}
+        _ ->
+          {:ok, nil}
+      end
```

Comment:
Because I love pattern matching on function heads:
```
fn
  _, %{source: source, definition: %{name: name}} when is_map(source) ->
    {:ok, Map.get(source, String.upcase(name))}

  _, _ ->
    {:ok, nil}
end
```